monocle/worker.py

```diff
@@ -780,8 +780,43 @@ def travel_speed(self, point):
                         if not cooldown or time() > cooldown / 1000:
                             spinning = LOOP.create_task(self.spin_pokestop(pokestop))
                 else:
-                    DB_PROC.add(self.normalize_gym(fort))
 
+                    fort['name'] = ""
+                    if not self.normalize_gym(fort) in FORT_CACHE:
+                        request = self.api.create_request()
+                        request.get_gym_details(gym_id=fort['id'],
+                                                player_latitude=latitude,
+                                                player_longitude=longitude,
+                                                gym_latitude=fort['latitude'],
+                                                gym_longitude=fort['longitude']
+                                                )
+                        responses = await self.call(request)
+
+
+
+                        result = responses.get('GET_GYM_DETAILS', {}).get('status', 0)
+                        if result == 0:
+                            try:
+                                get_gym_details = responses['GET_GYM_DETAILS']
+                                fort['name'] = get_gym_details['name']
+                                DB_PROC.add(self.normalize_gym(fort))
+                                for member in get_gym_details['gym_state']['memberships']:
+                                    rowDetail = {}
+                                    rowDetail['id'] = fort['id']
+                                    rowDetail['player_name'] = member['trainer_public_profile']['name']
+                                    rowDetail['player_level'] = member['trainer_public_profile']['level']
+                                    rowDetail['pokemon_id'] = member['pokemon_data']['pokemon_id']
+                                    rowDetail['pokemon_cp'] = member['pokemon_data']['cp']
+                                    rowDetail['last_modified_timestamp_ms'] = fort['last_modified_timestamp_ms']
+                                    DB_PROC.add(self.normalize_gym_detail(rowDetail))
```

Comment:
this should be `DB_PROC.add(self.normalize_gym_detail(fort, member))` deprecating the lines above